apps/mobile/src/components/add-wallet/add-wallet-sheet.tsx

```diff
@@ -83,63 +58,25 @@ function getUnavailableFeatures(theme: Theme) {
   return UNAVAILABLE_FEATURES;
 }
 
-interface AddWalletModalBaseProps {
-  addWalletModalRef: RefObject<BottomSheetModal>;
+interface AddWalletSheetBaseProps {
+  addWalletSheetRef: RefObject<SheetRef>;
 }
 
-export function AddWalletModal({ addWalletModalRef }: AddWalletModalBaseProps) {
-  const notifyUserModalRef = useRef<BottomSheetModal>(null);
-  const router = useRouter();
-  const [optionData, setOptionData] = useState<OptionData | null>(null);
-  const createWallet = useCallback(() => {
-    router.navigate(APP_ROUTES.WalletCreateNewWallet);
-    addWalletModalRef.current?.close();
-  }, [router]);
-
-  const restoreWallet = useCallback(() => {
-    router.navigate(APP_ROUTES.WalletRecoverWallet);
-    addWalletModalRef.current?.close();
-  }, [router]);
-
-  function onOpenNotificationsModal(option: OptionData) {
-    setOptionData(option);
-    notifyUserModalRef.current?.present();
-  }
-  function onCloseNotificationsModal() {
-    setOptionData(null);
-  }
-
-  return (
-    <>
-      <AddWalletModalUI
-        onOpenNotificationsModal={onOpenNotificationsModal}
-        createWallet={createWallet}
-        restoreWallet={restoreWallet}
-        addWalletModalRef={addWalletModalRef}
-      />
-
-      <NotifyUserModal
-        optionData={optionData}
-        onCloseNotificationsModal={onCloseNotificationsModal}
-        notifyUserModalRef={notifyUserModalRef}
-      />
-    </>
-  );
-}
-interface AddWalletModalUIProps extends AddWalletModalBaseProps {
+interface AddWalletSheetUIProps extends AddWalletSheetBaseProps {
   createWallet(): unknown;
   restoreWallet(): unknown;
   onOpenNotificationsModal(option: OptionData): unknown;
 }
-export function AddWalletModalUI({
-  addWalletModalRef,
+export function AddWalletSheetUI({
+  addWalletSheetRef,
   createWallet,
   restoreWallet,
   onOpenNotificationsModal,
-}: AddWalletModalUIProps) {
+}: AddWalletSheetUIProps) {
   const [moreOptionsVisible, setMoreOptionsVisible] = useState(false);
   const animatedIndex = useSharedValue<number>(CLOSED_ANIMATED_SHARED_VALUE);
   const theme = useTheme<Theme>();
+  const { theme: themeVariant } = useSettings();
```

Comment:
If this is a layout/style component, I think we should pass this in rather than use hook directly. Other than that, this is a clean abstraction

Reply:
Good catch 👍 I updated the naming to start using `.layout` and moved this hook into the container / smart component